File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville:list, K):
    heapq.heapify(scoville)
    result = 0
    while scoville[0] < K and len(scoville) > 1:
        curr = heapq.heappop(scoville) + heapq.heappop(scoville) * 2
        heapq.heappush(scoville, curr)
        result+=1
    if scoville[0] < K:
        return -1
    else:
        return result

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int result = 0;
        for(int i : scoville) {
            pq.offer(i);
        }
        while(pq.peek()<K) {
            Integer a = pq.poll();
            Integer b = pq.poll();
            if(a == null || a<=0 || b == null) {
                return -1;
            }
            pq.offer(a+b*2);
            result++;
        }

        return result;
    }
}